code/game/turfs/simulated/walls.dm

```diff
@@ -25,6 +25,13 @@
 	/turf/closed/wall/clockwork)
 	smooth = SMOOTH_TRUE
 
+/turf/closed/wall/examine(mob/user)
+	..()
+	deconstruction_hints(user)
```

Comment:
Would this be possible to be moved somewhere where examining comes from? So if user examines something which has deconstruction_hints, they would be shown automatically.

Reply:
you could move it up, and that's not a bad idea, you'd need to create default procs that have zero hints though at the turf level to match it and then override for your types with hints

Reply:
Ok, I figured out [this](https://github.com/tgstation/tgstation/blob/96718e9648935eae26cba8b89727751c2ddf4aec/code/game/atoms.dm#L267) is the examine I need to add it to, but I can't figure out how to make it call the deconstruction_hints-proc. Here's what I tried starting at L275:
```
...
	if(desc)
		to_chat(user, desc)

	var/deconstruction_hints
	if(deconstruction_hints)
		to_chat(user, deconstruction_hints)
...
```

Reply:
I'm not sure that turfs count as atoms, they're kinda snowflake

You might need to add it to both the atom and turf examine procs as well as add a default `atom/proc/deconstruction_hints` proc

Reply:
turfs are atoms, but atoms are not turfs

Reply:
Sorry but figuring this out is a little out of my reach right now. I'll do it the not-the-most-elegant way like it is right now, and maybe make this fine tune to it some other time, if that's ok?